pkg/operator/ceph/object/objectstore.go

```diff
@@ -88,25 +106,185 @@ func deleteRealmAndPools(context *Context, spec cephv1.ObjectStoreSpec) error {
-	updatePeriod := false
+	serviceEndpoint := fmt.Sprintf("http://%s:%d", serviceIP, spec.Gateway.Port)
+	if spec.Gateway.SecurePort != 0 && spec.Gateway.SSLCertificateRef != "" {
+		serviceEndpoint = fmt.Sprintf("https://%s:%d", serviceIP, spec.Gateway.SecurePort)
+	}
+
+	zoneEndpoints, err := getZoneEndpoints(context, realmArg, zoneGroupArg, zoneName, serviceEndpoint)
+	if err != nil {
+		return err
+	}
+	logger.Debugf("Endpoints for zone %q are: %q", zoneName, zoneEndpoints)
+	endpointArg := fmt.Sprintf("--endpoints=%s", zoneEndpoints)
 
 	if spec.IsMultisite() {
-		updatePeriod = true
-		_, err := RunAdminCommandNoRealm(context, "zonegroup", "modify", realmArg, zoneGroupArg, endpointArg)
+		zoneIsMaster, err := checkZoneIsMaster(context, realmArg, zoneGroupArg, zoneArg)
 		if err != nil {
-			return errors.Wrapf(err, "failed to add object store %q in rgw zone group %q", context.Name, zoneGroupName)
+			return err
+		}
+
+		if zoneIsMaster {
+			// endpoints that are part of a master zone are supposed to be the endpoints for a zone group
+			_, err := RunAdminCommandNoRealm(context, "zonegroup", "modify", realmArg, zoneGroupArg, endpointArg)
+			if err != nil {
+				return errors.Wrapf(err, "failed to add object store %q in rgw zone group %q", context.Name, zoneGroupName)
+			}
+			logger.Debugf("Added endpoints %q to zonegroup %q", zoneEndpoints, zoneGroupName)
 		}
 		_, err = RunAdminCommandNoRealm(context, "zone", "modify", realmArg, zoneGroupArg, zoneArg, endpointArg)
 		if err != nil {
 			return errors.Wrapf(err, "failed to add object store %q in rgw zone %q", context.Name, zoneName)
 		}
+
+		// the period will help notify other zones of changes if there are multi-zones
+		_, err = RunAdminCommandNoRealm(context, "period", "update", "--commit", realmArg, zoneGroupArg, zoneArg)
+		if err != nil {
+			return errors.Wrap(err, "failed to update period")
+		}
 		logger.Infof("Added object store %q to realm %q, zonegroup %q, zone %q", context.Name, realmName, zoneGroupName, zoneName)
+
+		// create system user for realm for master zone in master zonegorup for multisite scenario
+		zoneGroupIsMaster, err := checkZoneGroupIsMaster(context, realmArg, zoneGroupArg)
+		if zoneIsMaster && zoneGroupIsMaster {
+			uid := realmName + "-system-user"
+			uidArg := fmt.Sprintf("--uid=%s", uid)
+			output, err := RunAdminCommandNoRealm(context, "user", "info", uidArg)
+			if err != nil {
+				if code, ok := exec.ExitStatus(err); ok && code == int(syscall.EINVAL) {
+					logger.Debugf("realm system user %q not found, running `radosgw-admin user create`", uid)
+					accessKeyArg, secretKeyArg, err := GetRealmKeyArgs(context.Context, realmName, namespace)
+					if err != nil {
+						return errors.Wrap(err, "failed to get keys for realm")
+					}
+					systemArg := "--system"
+					displayName := realmName + ".user"
+					displayNameArg := fmt.Sprintf("--display-name=%s", displayName)
```

Comment:
No need to define `displayName` on the previous line
```suggestion
					displayNameArg := fmt.Sprintf("--display-name=%s.user", realmName)
```